testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -552,6 +553,71 @@ public void testJoinBaseTableWithView()
         assertAccessAllowed(viewOwnerSession, "DROP VIEW " + viewName);
     }
 
+    @Test
+    public void testFunctionOwners()
+    {
+        reset();
+
+        String functionOwner = "function_owner";
+        TrinoPrincipal functionOwnerPrincipal = new TrinoPrincipal(USER, functionOwner);
+
+        systemSecurityMetadata.grantRoles(getSession(), ImmutableSet.of("function_owner_role"), ImmutableSet.of(functionOwnerPrincipal), false, Optional.empty());
+        systemSecurityMetadata.setFunctionOwner(
+                getSession(),
+                new CatalogSchemaFunctionName("memory", "default", "my_test_function_outer"),
+                functionOwnerPrincipal);
+
+        Session functionOwnerSession = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.forUser(functionOwner)
+                        .withEnabledRoles(Set.of("function_owner_role"))
+                        .build())
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+        assertAccessAllowed(
+                functionOwnerSession,
+                "CREATE FUNCTION memory.default.my_test_function_inner (x integer) RETURNS bigint RETURN x + 42");
+        assertAccessAllowed(
```

Comment:
How about a check with the default session?

Reply:
What would be the point of such a test?  Verification if a function can be created is out of scope for this commit